paimon-common/src/main/java/org/apache/paimon/data/columnar/ColumnarRow.java

```diff
@@ -221,4 +225,18 @@ public int hashCode() {
         throw new UnsupportedOperationException(
                 "ColumnarRowData do not support hashCode, please hash fields one by one!");
     }
+
+    public void mapping(@Nullable PartitionInfo partitionInfo, @Nullable int[] indexMapping) {
```

Comment:
Return a new `ColumnarRow`.
If there is no performance issue, it is better to keep objects imummable.

Reply:
Fixed this.